README.rst

```diff
@@ -13,146 +13,143 @@ smart_open -- utils for streaming large files
+  >>> # stream from/to local compressed files:
+  >>> for line in smart_open('./foo.txt.gz'):
   ...    print line
 
-  >>> # you can also use a boto.s3.key.Key instance directly:
-  >>> key = boto.connect_s3().get_bucket("my_bucket").get_key("my_key")
-  >>> with smart_open.smart_open(key) as fin:
-  ...     for line in fin:
-  ...         print line
+  >>> with smart_open('/home/radim/foo.txt.bz2', 'wb') as fout:
+  ...    fout.write("some content\n")
 
   >>> # can use context managers too:
-  >>> with smart_open.smart_open('s3://mybucket/mykey.txt') as fin:
+  >>> with smart_open('s3://mybucket/mykey.txt') as fin:
   ...     for line in fin:
   ...         print line
   ...     fin.seek(0)  # seek to the beginning
   ...     print fin.read(1000)  # read 1000 bytes
 
   >>> # stream from HDFS
-  >>> for line in smart_open.smart_open('hdfs://user/hadoop/my_file.txt'):
+  >>> for line in smart_open('hdfs://user/hadoop/my_file.txt'):
   ...     print line
 
   >>> # stream from HTTP
-  >>> for line in smart_open.smart_open('http://example.com/index.html'):
+  >>> for line in smart_open('http://example.com/index.html'):
   ...     print line
 
   >>> # stream from WebHDFS
-  >>> for line in smart_open.smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
+  >>> for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
   ...     print line
 
   >>> # stream content *into* S3 (write mode):
-  >>> with smart_open.smart_open('s3://mybucket/mykey.txt', 'wb') as fout:
+  >>> with smart_open('s3://mybucket/mykey.txt', 'wb') as fout:
   ...     for line in ['first line', 'second line', 'third line']:
   ...          fout.write(line + '\n')
 
   >>> # stream content *into* HDFS (write mode):
-  >>> with smart_open.smart_open('hdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
+  >>> with smart_open('hdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
   ...     for line in ['first line', 'second line', 'third line']:
   ...          fout.write(line + '\n')
 
   >>> # stream content *into* WebHDFS (write mode):
-  >>> with smart_open.smart_open('webhdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
+  >>> with smart_open('webhdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
   ...     for line in ['first line', 'second line', 'third line']:
   ...          fout.write(line + '\n')
 
-  >>> # stream from/to local compressed files:
-  >>> for line in smart_open.smart_open('./foo.txt.gz'):
+  >>> # stream using a completely custom s3 server, like s3proxy:
+  >>> for line in smart_open('s3u://user:secret@host:port@mybucket/mykey.txt'):
   ...    print line
 
-  >>> with smart_open.smart_open('/home/radim/foo.txt.bz2', 'wb') as fout:
-  ...    fout.write("some content\n")
+  >>> # you can also use a boto.s3.key.Key instance directly:
+  >>> key = boto.connect_s3().get_bucket("my_bucket").get_key("my_key")
+  >>> with smart_open(key) as fin:
+  ...     for line in fin:
+  ...         print line
```

Comment:
Should we use `print(line)` instead of `print line`? The former works under both Py2 and Py3, the latter doesn't.

Reply:
Yes! Let me fix that.